src/transforms/javascript.rs

```diff
@@ -0,0 +1,789 @@
+use super::Transform;
+use crate::{
+    event::{Event, ValueKind},
+    topology::config::{DataType, TransformConfig},
+};
+use lazy_static::lazy_static;
+use quick_js::{Context, JsValue};
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::{HashMap, HashSet},
+    fs,
+    sync::mpsc,
+    thread,
+};
+
+lazy_static! {
+    // although JavaScript identifiers can also contain Unicode characters, we don't allow them
+    static ref JS_VALID_IDENTIFIER: Regex = Regex::new(r"^[a-zA-Z_$][a-zA-Z0-9_$]*$").unwrap();
+    static ref JS_RESERVED_KEYWORDS: HashSet<&'static str> = [
+        // reserved keywords
+        "break",
+        "case",
+        "catch",
+        "class",
+        "const",
+        "continue",
+        "debugger",
+        "default",
+        "delete",
+        "do",
+        "else",
+        "export",
+        "extends",
+        "finally",
+        "for",
+        "function",
+        "if",
+        "import",
+        "in",
+        "instanceof",
+        "new",
+        "return",
+        "super",
+        "switch",
+        "this",
+        "throw",
+        "try",
+        "typeof",
+        "var",
+        "void",
+        "while",
+        "with",
+        "yield",
+        // future reserved keyword
+        "enum",
+        // literals that cannot be used as identifiers
+        "null",
+        "true",
+        "false"
+    ]
+    .iter()
+    .cloned()
+    .collect();
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct JavaScriptConfig {
```

Comment:
There is no single required config item because either `source` or `path` could be used to create a script.

Reply:
> Is it we need either source or path but if neither are there we fail?
We fail if there are no source or path and also fail if there are both source and path.

The case of `RegionOrEndpoint` looks similar, so similar approach can be applied here.

I think using [untagged](https://serde.rs/enum-representations.html#untagged) representation for new `SourceOrPath` struct can make it possible to avoid explicit matching at all, expressing everything just using Serde.

Reply:
Yeah, this is my preferred method.